packages/calypso-e2e/src/lib/utils/tracks-event-manager.ts

```diff
@@ -0,0 +1,125 @@
+import { Page, Request } from 'playwright';
+
+// Modify global Window interface to include _tkAllowE2ETests
+declare global {
+	interface Window {
+		_tkAllowE2ETests: boolean;
+	}
+}
+
+/**
+ * A class to help monitor Tracks events in the browser
+ */
+export class TracksEventManager {
+	private page: Page;
+	private timeout: number;
+
+	/**
+	 * Construct an instance of the Tracks event manager class
+	 * @param page
+	 * @param timeout
+	 */
+	constructor( page: Page, timeout: number = 10000 ) {
+		this.page = page;
+		this.timeout = timeout;
+		this.blockUnnecessaryRequests();
+	}
+
+	/**
+	 * Tell Tracks to allow these tests to fire events
+	 * We later abort any requests to t.gif
+	 */
+	async allowTestsToFireEvents() {
+		await this.page.addInitScript( () => {
+			window._tkAllowE2ETests = true;
+		} );
+	}
+
+	/**
+	 * Navigate to a URL
+	 * @param url
+	 */
+	async navigateToUrl( url: string ) {
+		try {
+			await this.page.goto( url, { timeout: this.timeout } );
+		} catch ( error ) {
+			console.error( error );
+		}
+	}
+
+	/**
+	 * Check if a Tracks event fired
+	 * @param eventName
+	 * @returns
+	 */
+	async didEventFire( eventName: string ) {
+		return !! ( await this.getRequestUrlForEvent( eventName ) );
+	}
+
+	/**
+	 * Get the request URL of a Tracks event
+	 * @param eventName
+	 * @returns
+	 */
+	async getRequestUrlForEvent( eventName: string ): Promise< string > {
+		const pixelUrl: string = 'https://pixel.wp.com/t.gif';
+		return new Promise( ( resolve ) => {
+			const timeoutId = setTimeout( () => {
+				resolve( '' );
+			}, this.timeout );
+
+			const handler = ( request: Request ) => {
+				if (
+					request.url().startsWith( pixelUrl ) &&
+					this.getParamFromUrl( '_en', request.url() ) === eventName
+				) {
+					this.page.removeListener( 'request', handler );
+					clearTimeout( timeoutId );
+					resolve( request.url() );
+				}
+			};
+
+			this.page.on( 'request', handler );
```

Comment:
Can we use [waitForRequest](https://playwright.dev/docs/api/class-page#page-wait-for-request) here? It should allow us to drop the custom timeout handler. For example:

```js
async getRequestUrlForEvent( eventName: string ) {
  const request = await this.page.waitForRequest( ( request ) => (
    request.url().startsWith( pixelUrl ) &&
    this.getParamFromUrl( '_en', request.url() ) === eventName
  ), { timeout: this.timeout } );

  return request.url();
}
```

Reply:
Ah, very nice! I'll make that change. Thanks.